### PR TITLE
SaveAs tool and example

### PR DESCRIPTION
**Pull request by swederik**
Small function for saving PNG files from the canvases. Addresses #31.

~~Seems it can't be automatically merged because of changes to dist/cornerstoneTools... How should we approach this? I assume I stop using grunt before commits? But then the rawgit examples won't work...~~

Update: Rebased and fixed

saveas

### CODE REVIEW

**Comment by cancan101** [comment]
Is it possible to use any of this code for send the binary image over HTTP, for example using [`window.fetch`](https://github.com/github/fetch#file-upload)?

**Comment by swederik** [comment]
Yeah, sure, but what for? You want to save the canvas as an image on your server?

I think you can just use canvas.toDataURL(); and then POST that data to the server. Then you just decode the base64 string to produce an image on the server.

**Comment by cancan101** [comment]
Yes, exactly. I want to save the save the canvas (which has all of the user's modifications, transforms and annotations) to the server. This along with #30 so that I have both the steps to re-create the modifications and the image of the canvas.

**Comment by swederik** [comment]
Ok so you want to use the image to show the saved-state then. Seems easy enough to do, but I don't know if it's worth having extra code for it in cornerstone itself.

```` javascript
// Get the canavas
var canvas = $(element).find("canvas").get(0);
var imageData = canvas.toDataURL();
$.ajax({
  method: "POST",
  url: "/your/server/",
  data: {"image": imageData}
})
````

Then on the server (at least in Python):

```` python
import base64

// Assuming you have a request like in Django
data = request.POST["image"]

filename = "MyNewImage.png"
out_file = open(filename, "w")
out_file.write(base64.decodestring(data))
out_file.close()
````

Hope that helps

**Comment by cancan101** [comment]
Sure that seems simple enough. I do think though that a save as tool for in browser saves still stands on its own as a useful feature.

File: dist/cornerstoneTools.js [hunk 1]
```diff
@@ -2229,6 +2229,55 @@ var cornerstoneTools = (function ($, cornerstone, cornerstoneMath, cornerstoneTo
...
+
+    "use strict";
+
+    if(cornerstoneTools === undefined) {
+        cornerstoneTools = {};
+    }
+
+    function saveAs(element, filename)
+    {
+        var canvas = $(element).find("canvas").get(0);
```
Comment (cancan101) [inline dist/cornerstoneTools.js]: Does it make sense to extract `$(element).find("canvas").get(0);` as a helper method?

Comment (swederik) [inline dist/cornerstoneTools.js]: Probably not, it's boilerplate jQuery.

**Pull request merged**